### PR TITLE
Tool for creating room events

### PR DESCRIPTION
**Pull request by NegativeMjark**
I'm starting a collection of tools under `matrix-org/dendrite/tools`. The first is a tool for generating lots of room events for testing purposes.

### CODE REVIEW

**Comment by Kegsay** [comment]
Tools need usage and help messages on the command prompt. This does not appear to have that. Please add (I appreciate you do have comments on things, but that's not a great help for the user).

File: src/github.com/matrix-org/dendrite/tools/create-room-events/main.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		panic(err)
+	}
+	writeEvent(event)
+	reference := event.EventReference()
+	b.PrevEvents = []gomatrixserverlib.EventReference{reference}
+	b.Depth++
+	return reference
+}
+
+// Write an event to the output.
```
Comment (Kegsay) [inline src/github.com/matrix-org/dendrite/tools/create-room-events/main.go]: one event per-line? `\n` delimited?

File: src/github.com/matrix-org/dendrite/tools/create-room-events/main.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	b.AuthEvents = []gomatrixserverlib.EventReference{create}
+	member := build()
+
+	// Build a number of m.room.message events.
+	b.Type = "m.room.message"
+	b.StateKey = nil
+	b.SetContent(map[string]string{"body": "Test Message"})
+	b.AuthEvents = []gomatrixserverlib.EventReference{create, member}
+	for i := 0; i < count; i++ {
+		build()
```
Comment (Kegsay) [inline src/github.com/matrix-org/dendrite/tools/create-room-events/main.go]: Unhelpful name given it also writes to stdout. `buildAndOutput` is a bit more mouthy but is a lot easier to follow.

File: src/github.com/matrix-org/dendrite/tools/create-room-events/main.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	privateKeyString = defaulting(os.Getenv("PRIVATE_KEY"), defaultKey)
+	// The room ID to generate events in.
+	roomID = defaulting(os.Getenv("ROOM_ID"), "!roomid:"+serverName)
+	// The user ID to use for as the event sender.
+	userID = defaulting(os.Getenv("USER_ID"), "@userid:"+serverName)
+	// The number of m.room.messsage events to generate.
+	messageCount = defaulting(os.Getenv("MESSAGE_COUNT"), "10")
+	// The output format to use for the messages.
+	//	INPUT -> api.InputRoomEvent
+	//  RAW   -> gomatrixserverlib.Event
```
Comment (Kegsay) [inline src/github.com/matrix-org/dendrite/tools/create-room-events/main.go]: Adding more terminology for a dendrite-only tool feels sadness making to me. Can we not just use `Event` and `InputRoomEvent` (the actual type names)?

Also, tabs!

**Comment by NegativeMjark** [comment]
> I would like to know why we are using env vars rather than https://golang.org/pkg/flag/ though.

No particular reason, it makes implementing the defaults a bit easier. (changing SERVER_NAME changes the defaults for the other variables). But I could probably have done it with the flags module. if necessary.

**Comment by Kegsay** [comment]
LGTM - We may have people whine about it being env vars though, since it is an unusual choice over `--flags`.

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**